### DISCUSSION TITLE
Just an FYI for Katex integration

### DISCUSSION THREAD

**Discussion by bilal-fazlani** [squidfunk/mkdocs-material#6807] [_]
Just an FYI, if you are using [math features](https://squidfunk.github.io/mkdocs-material/reference/math) with Katex and wondering why its not working with instant navigation, 

try below script instead of the [this one](https://squidfunk.github.io/mkdocs-material/reference/math/#katex)

```js
document$.subscribe(() => {
  renderMathInElement(document.body);
});
```

the only difference is it uses `document.body` global variable instead the function variable `body`. After doing this math will start rendering when navigating pages.

**Comment by Burhan-Q** [comment]
Can confirm this works. I found that on visiting a page, KaTeX wouldn't render, but it _would_ when refreshed. Making this change fixed the rendering for the page load (locally).

**Comment by kamilkrzyskow** [comment]
There is an open PR for the docs change:
- https://github.com/squidfunk/mkdocs-material/pull/6823

but it would be nice to know what has changed that would require setting the document.body 🤔

**Reply by squidfunk** [reply]
Sounds like a potential regression. I'll look into it.

**Comment by squidfunk** [comment]
It turned out to be a regression of [9.5.5](https://github.com/squidfunk/mkdocs-material/releases/tag/9.5.5), which was fixed in de46c41.

**Reply by squidfunk** [reply]
Released as part of [9.5.12](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.12).
